src/compiler/parser.ts

```diff
@@ -1111,6 +1111,14 @@ namespace ts {
             return token === t && tryParse(nextTokenCanFollowModifier);
         }
 
+        function nextTokenCanFollowModifierWithoutGrammarRules() {

```

Comment:
@CyrusNajmabadi I still struggle with coming up with good names. I just needed to refactor some code to a method to prevent code duplication.

@DanielRosenwasser Thank you